core/src/main/java/org/kiwix/kiwixmobile/core/history/HistoryActivity.kt

```diff
@@ -0,0 +1,173 @@
+import kotlinx.android.synthetic.main.activity_history.recycler_view
+import kotlinx.android.synthetic.main.layout_toolbar.toolbar
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.R.id
+import org.kiwix.kiwixmobile.core.R.string
+import org.kiwix.kiwixmobile.core.base.BaseActivity
+import org.kiwix.kiwixmobile.core.di.components.CoreComponent
+import org.kiwix.kiwixmobile.core.extensions.ActivityExtensions.coreActivityComponent
+import org.kiwix.kiwixmobile.core.extensions.ActivityExtensions.viewModel
+import org.kiwix.kiwixmobile.core.history.adapter.HistoryAdapter
+import org.kiwix.kiwixmobile.core.history.adapter.HistoryAdapter.OnItemClickListener
+import org.kiwix.kiwixmobile.core.history.adapter.HistoryDelegate.HistoryDateDelegate
+import org.kiwix.kiwixmobile.core.history.adapter.HistoryDelegate.HistoryItemDelegate
+import org.kiwix.kiwixmobile.core.history.adapter.HistoryListItem.HistoryItem
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action.ExitHistory
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action.Filter
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action.OnItemLongClick
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action.RequestDeleteAllHistoryItems
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action.RequestDeleteSelectedHistoryItems
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action.ToggleShowHistoryFromAllBooks
+import org.kiwix.kiwixmobile.core.history.viewmodel.HistoryViewModel
+import org.kiwix.kiwixmobile.core.history.viewmodel.State
+import org.kiwix.kiwixmobile.core.history.viewmodel.State.NoResults
+import org.kiwix.kiwixmobile.core.history.viewmodel.State.Results
+import org.kiwix.kiwixmobile.core.history.viewmodel.State.SelectionResults
+import org.kiwix.kiwixmobile.core.utils.DialogShower
+import org.kiwix.kiwixmobile.core.utils.SimpleTextListener
+import javax.inject.Inject
+
+const val USER_CLEARED_HISTORY: String = "user_cleared_history"
+
+class HistoryActivity : OnItemClickListener, BaseActivity() {
+  private val activityComponent by lazy { coreActivityComponent }
+  private var actionMode: ActionMode? = null
+  @Inject lateinit var viewModelFactory: ViewModelProvider.Factory
+  private val historyViewModel by lazy { viewModel<HistoryViewModel>(viewModelFactory) }
+  private val compositeDisposable = CompositeDisposable()
+
+  @Inject lateinit var dialogShower: DialogShower
+
+  private val actionModeCallback: Callback =
+    object : Callback {
+      override fun onCreateActionMode(mode: ActionMode, menu: Menu): Boolean {
+        mode.menuInflater.inflate(R.menu.menu_context_delete, menu)
+        return true
+      }
+
+      override fun onPrepareActionMode(mode: ActionMode, menu: Menu): Boolean {
+        return false
+      }
+
+      override fun onActionItemClicked(mode: ActionMode, item: MenuItem): Boolean {
+        if (item.itemId == id.menu_context_delete) {
```

Comment:
make sure you always have full imports for R.* things. It can mess up the lint/inspection reports